crates/dbsp/src/trace/layers/column_layer/mod.rs

```diff
@@ -148,6 +161,9 @@ impl<K, R> ColumnLayer<K, R> {
         // since we may make some holes during the process.
         unsafe { self.set_len(0) };
 
+        let lower_bound = self.lower_bound;
+        self.lower_bound = 0;
```

Comment:
Around here we should drop all items from `0..lower_bound` if those items are useless to us